pydantic/types.py

```diff
@@ -497,3 +503,46 @@ def validate(cls, v: str) -> Any:
             raise errors.JsonError()
         except TypeError:
             raise errors.JsonTypeError()
+
+
+class IPAddress(str):
+    """
+    Type fully compatible with ipaddress.IPAddress from the stdlib.
+
+    See more:
+    https://docs.python.org/library/ipaddress.html
+    """
+
+    _constructor = ipaddress.ip_address
+    _exc = errors.IPAddressError
+
+    @classmethod
+    def __get_validators__(cls) -> 'CallableGenerator':
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value: IPAddr) -> str:
+        try:
+            ip = cls._constructor(value)
+        except ValueError:
+            raise cls._exc()
+        else:
+            return str(ip)
+
+
+class IPv4Address(IPAddress):
```

Comment:
Hi, I'm afraid this isn't the right way of implementing IP address verification.

The point is that the stdlib version of [`IPv4Address`](https://docs.python.org/3/library/ipaddress.html#ipaddress.IPv4Address) and `IPv6Address` should work as type annotations.

The only custom class we should implement is `IPvAnyAddress` which can be either a v4 or v6 address, as per #333 this should inherit from `_BaseAddress`, but `validate` should return either a `IPv4Address` or `IPv6Address`.

Have a look at [this code](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/validators.py#L267-L287) for how we validate standard lib types.

Reply:
Okay, now I see what's should be done.

But shall the ``validate`` take in either ``str``, ``bytes`` or ``int`` just like ``ipaddress.ip_address`` does?

Reply:
yes, argument annotation should be `Union[str, bytes, int]`

Reply:
I've just submitted a new implementation.